src/app/shared/components/validation-hint/validation-hint.component.ts

```diff
@@ -58,6 +60,7 @@ export class ValidationHintComponent implements OnInit, OnDestroy, OnChanges {
         Object.keys(this.validationFormControl.controls).forEach((key) => {
           this.updateValidationState(this.validationFormControl.get(key) as FormControl);
         });
+        this.checkFormLevelValidationErrors(this.validationFormControl.errors);
```

Comment:
can we use `this.updateValidationState(this.validationFormControl)` instead?

Reply:
It is to handle invalidTimeRange error. I set it to formControls (startTime and endTime) to highlight the input areas and i set it to formGroup to display validation hint. If I were to handle this error via this.updateValidationState(this.validationFormControl) , validation hint would be displayed twice for start time and end time. By using the form group level validation, the validation hint is displayed only once for the entire form.